txnkv/txnlock/lock_resolver.go

```diff
@@ -113,17 +113,27 @@ func (lr *LockResolver) Close() {
 
 // TxnStatus represents a txn's final status. It should be Lock or Commit or Rollback.
 type TxnStatus struct {
+	// The ttl is set from the `CheckTxnStatus` kv response, it is read only and do not change it.
 	ttl         uint64
 	commitTS    uint64
 	action      kvrpcpb.Action
 	primaryLock *kvrpcpb.LockInfo
 }
 
 // IsCommitted returns true if the txn's final status is Commit.
-func (s TxnStatus) IsCommitted() bool { return s.ttl == 0 && s.commitTS > 0 }
+func (s TxnStatus) IsCommitted() bool { return s.commitTS > 0 }
 
 // IsRolledBack returns true if the txn's final status is rolled back.
-func (s TxnStatus) IsRolledBack() bool { return s.ttl == 0 && s.commitTS == 0 }
+func (s TxnStatus) IsRolledBack() bool {
+	return s.ttl == 0 && s.commitTS == 0 && (s.action == kvrpcpb.Action_NoAction ||
+		s.action == kvrpcpb.Action_LockNotExistRollback ||
+		s.action == kvrpcpb.Action_TTLExpireRollback)
```

Comment:
I also asked in the original PR. Should `Action_TTLExpirePessimisticRollback` and `Action_LockNotExistDoNothing` be considered rolled back?

Reply:
Action_TTLExpirePessimisticRollback just pessimistic rollback the lock,
Action_LockNotExistDoNothing does not write a rollback record and it is returned when a pessimistic lock is encounttred(`resolving_pessimistic_lock == true`), 

So the transaction status from these checking results is undetermined, should not be regarded as already rollback